src/main/java/dev/dbos/transact/config/DBOSConfig.java

```diff
@@ -22,15 +20,28 @@ public record DBOSConfig(
     String appVersion,
     String executorId) {
 
-  public DBOSConfig {
-    Objects.requireNonNull(appName, "DBOSConfig.appName must not be null");
-    Objects.requireNonNull(databaseUrl, "DBOSConfig.databaseUrl must not be null");
-    Objects.requireNonNull(dbUser, "DBOSConfig.dbUser must not be null");
-    Objects.requireNonNull(dbPassword, "DBOSConfig.dbPassword must not be null");
-  }
-
   private static final Logger logger = LoggerFactory.getLogger(DBOSConfig.class);
 
+  public static DBOSConfig defaultsFromEnv(String appName) {
```

Comment:
This is nice and concise, but hides too much, I think it's better our sample code be explicit so users know how to change things later (and also so it's clearer what the env vars actually are).